internal/background/availability_queue.go

```diff
@@ -31,16 +42,40 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 	logger := ctxval.Logger(ctx)
 	ticker := time.NewTicker(tickDuration)
 	messageBuffer := make([]*kafka.GenericMessage, 0, batchSize)
+	sc, err := clients.GetSourcesClient(ctx)
+	if err != nil {
+		logger.Warn().Msg("Could not get sources client")
+		return
+	}
```

Comment:
Why are we performing any sort of work in the API? This can be done in the background statuser without putting any stress on our API pods. Kafka can handle twice or three times of message every hour, that is exactly why Kafka was designed for - bunch of messages sent or streamed.

This is actually serious design problem because now we have a goroutine running in each API worker pod that is now responsible from doing some REST calls. Because it is just one goroutine (and HTTP handlers spawn new goroutine for each request), it can be overloaded and lagging behind. So we should probably start monitoring the lag and errors (what if sources is down). And this is a chicken and egg problem.

Please can we keep this simple so the API workers just enqueue all checks via Kafka without doing any extra logic. I think the design is clear here - we want to do all the work in statuser so we can throttle it down when we hit any performance or AWS API limits issues.